File: sapphire/examples/kvstore/src/main/js/sapphire/demo/KeyValueStore.js
/**
 * A Key-Value Store where keys are {@code String}s and
 * values are {@code Serializable}s.
 */
class JSKeyValueStore {
    constructor() {
        this.list = new Map()
    }

    get(key){
        console.log(`server: getting value with key : ${key}`)
        return this.list.get(key)
    }

    set(key, value) {
        console.log(`server: setting ${key} = ${value}`)
        this.list.set(key,value)
    }

    contains(key) {
        console.log(`server: checking existence with key: ${key}`)
        return this.list.has(key)
    }
}

/*
// For test purpose only
function test() {
    kvs = new JSKeyValueStore()

    for (i=0; i<30; ++i) {
        key = "key_" + i;
        val = "val_" + i;

        console.log(`client: setting ${key} = ${val}`)
        kvs.set(key, val)

        var v = kvs.get(key)
        console.log(`client: got value ${v} with key ${key}`)
    }
}

test();
*/

File: sapphire/examples/kvstore/src/main/java/amino/run/demo/KeyValueStore.java
package amino.run.demo;

import amino.run.app.MicroService;

import java.io.Serializable;
import java.util.Hashtable;
import java.util.Map;

/**
 * A Key-Value Store where keys are {@code String}s and
 * values are {@code Serializable}s.
 */
public class KeyValueStore implements MicroService {
    private Map<String, Serializable> kvStore = new Hashtable<>();

    public KeyValueStore() {}

    public void set(String key, Serializable value) {

        System.out.println(String.format("<Server>: setting %s = %s", key, value));
        this.kvStore.put(key, value);
    }

    public Serializable get(String key) {
        System.out.println(String.format("<Server>: getting value with key: %s", key));
        return this.kvStore.get(key);
    }

    public Boolean contains(String key) {
        System.out.println(String.format("<Server>: checking existence with key: %s", key));
        return this.kvStore.containsKey(key);
    }
}
